spec/support/corpus.rb

```diff
@@ -226,6 +243,53 @@ def system(arguments)
         end
       end
 
+      # Mapping of files which we expect to cause errors during mutation generation
+      class ErrorWhitelist
+        UnnecessaryExpectation = Class.new(StandardError)
+        UNNECESSARY_EXPECTATION = 'Expected to encounter %s while mutating "%s"'.freeze
```

Comment:
Instead of defining this as a constant and having it loosely paired with the `UnnecessaryExpectation` constant, what about defining an actual `UnnecessaryExpectation` class where this is the default message? Then when you raise the exception below you could use something like:

```ruby
fail UnnecessaryExpectation.new(map, path)
```